### PR TITLE
Adding initial recipe for ncagg

### PR DESCRIPTION
**Pull request by 5tefan**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by 5tefan** [comment]
@conda-forge/help-python, I think this is ready. Thank you!

**Comment by 5tefan** [comment]
@conda-forge-admin, please ping team

File: recipes/ncagg/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+requirements:
+  host:
+    - python >=3.7
+    - pip
+  run:
+    - python >=3.7
+    - numpy
+    - netCDF4
+    - Cerberus >=1.3,<1.4
+    - Click >=7.1,<7.2
```
Comment (xhochy) [inline recipes/ncagg/meta.yaml:27]: These package names are lowercase in conda-forge
```suggestion
    - netcdf4
    - cerberus >=1.3,<1.4
    - click >=7.1,<7.2
```

**Comment by 5tefan** [comment]
Thanks. Ahh, sorry, I just noticed "Ran 0 tests in 0.000s" [1] in the log. There should be some tests. Do I need to change the test command to something like `python -m unittest discover -s $SRC_DIR`? (I think I will try this)


[1] https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=300034&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=765

File: recipes/ncagg/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+source:
+  url: https://github.com/5tefan/ncagg/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 49517f3bc6d3e6c2e476dda5e38a1646e331ea12a07eef049084faae0dc6f651
+
+build:
+  noarch: python
+  entry_points:
+    - "ncagg = ncagg.cli:cli"
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (xhochy) [inline recipes/ncagg/meta.yaml:16]: You can run the tests as follows during the build.
```suggestion
  script: 
    - "{{ PYTHON }} -m pip install . -vv"
    - python -m unittest discover -s $SRC_DIR
```

File: recipes/ncagg/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    - numpy
+    - netcdf4
+    - cerberus >=1.3,<1.4
+    - click >=7.1,<7.2
+
+test:
+  imports:
+    - ncagg
+  commands:
+    - python -m unittest discover -s .
```
Comment (xhochy) [inline recipes/ncagg/meta.yaml:33]: During test we though mostly do import and requirements consistency checks. Unit tests are nice to have but not required.
```suggestion
test:
  imports:
    - ncagg
  requires:
    - pip
  commands:
    - pip check
```

File: recipes/ncagg/meta.yaml (hunk 2 above)
Comment (5tefan) [reply recipes/ncagg/meta.yaml:16]: Ah, interesting. Thanks!

**Comment by 5tefan** [comment]
@conda-forge-admin, please restart ci

**Comment by xhochy** [comment]
This is ready to merge. We can ignore the Windows failure as this recipe will be build on Linux only on the final feedstock. @5tefan if you are happy with the current state, I would push the merge button.

**Pull request merged**